code/modules/hydroponics/hydroponics.dm

```diff
@@ -521,6 +521,12 @@
 		mutmod = 0
 		adjustNutri(round(S.get_reagent_amount("robustharvestnutriment") *1 ))
 
+
+	//Fish Water is both an excellent fertilizer and waters
+	if(S.has_reagent("fishwater", 1))
```

Comment:
It's not great, no; both botany systems had this type of horrendous if chain to handle reagents, though there's far more reagents, with this system that interact with plants. I disliked it greatly, but I wasn't about to do a refactor of a refactor amidst 230-some file changes.